### PR TITLE
Optimize ByteArrayScalarConversionPolicy

### PR DESCRIPTION
**Pull request by sungam3r**

### CODE REVIEW

File: src/Serilog/Policies/ByteArrayScalarConversionPolicy.cs [hunk 1]
```diff
@@ -31,13 +31,22 @@ public bool TryConvertToScalar(object value, [NotNullWhen(true)] out ScalarValue
...
+            var start = Convert.ToHexString(bytes, 0, 16);
+#else
             var start = string.Concat(bytes.Take(16).Select(b => b.ToString("X2")));
+#endif
             var description = start + "... (" + bytes.Length + " bytes)";
             result = new ScalarValue(description);
         }
         else
         {
+#if NET5_0_OR_GREATER
```
Comment (SimonCropp) [inline src/Serilog/Policies/ByteArrayScalarConversionPolicy.cs:45]: perhaps add a new constant `FEATURE_TOHEXSTRING` ?

Comment (sungam3r) [reply src/Serilog/Policies/ByteArrayScalarConversionPolicy.cs]: done

File: src/Serilog/Serilog.csproj [hunk 2]
```diff
@@ -32,15 +32,15 @@
...
   </PropertyGroup>
 
   <PropertyGroup Condition=" '$(TargetFramework)' == 'net6.0' ">
-    <DefineConstants>$(DefineConstants);FEATURE_ASYNCLOCAL;FEATURE_HASHTABLE;FEATURE_DEFAULT_INTERFACE;FEATURE_SPAN;FEATURE_DATE_AND_TIME_ONLY;FEATURE_VALUETUPLE;FEATURE_ITUPLE;FEATURE_ASYNCDISPOSABLE;FEATURE_WRITE_STRINGBUILDER</DefineConstants>
+    <DefineConstants>$(DefineConstants);FEATURE_ASYNCLOCAL;FEATURE_HASHTABLE;FEATURE_DEFAULT_INTERFACE;FEATURE_SPAN;FEATURE_DATE_AND_TIME_ONLY;FEATURE_VALUETUPLE;FEATURE_ITUPLE;FEATURE_ASYNCDISPOSABLE;FEATURE_WRITE_STRINGBUILDER;FEATURE_TOHEXSTRING</DefineConstants>
   </PropertyGroup>
 
   <PropertyGroup Condition=" '$(TargetFramework)' == 'net7.0' ">
-    <DefineConstants>$(DefineConstants);FEATURE_ASYNCLOCAL;FEATURE_HASHTABLE;FEATURE_DEFAULT_INTERFACE;FEATURE_SPAN;FEATURE_DATE_AND_TIME_ONLY;FEATURE_VALUETUPLE;FEATURE_ITUPLE;FEATURE_ASYNCDISPOSABLE</DefineConstants>
+    <DefineConstants>$(DefineConstants);FEATURE_ASYNCLOCAL;FEATURE_HASHTABLE;FEATURE_DEFAULT_INTERFACE;FEATURE_SPAN;FEATURE_DATE_AND_TIME_ONLY;FEATURE_VALUETUPLE;FEATURE_ITUPLE;FEATURE_ASYNCDISPOSABLE;FEATURE_WRITE_STRINGBUILDER;FEATURE_TOHEXSTRING</DefineConstants>
```
Comment (sungam3r) [inline src/Serilog/Serilog.csproj:43]: I added FEATURE_WRITE_STRINGBUILDER here because it was lost in one of previous PR.

**Comment by sungam3r** [comment]
Conflicts resolved.

**Pull request merged**